firmware/net/components/wifi/wifi.cc

```diff
@@ -26,125 +31,168 @@ Wifi::~Wifi()
 
 void Wifi::Begin()
 {
-    connect_semaphore = xSemaphoreCreateBinary();
-
-    // Start with a value of 1 for readability
-    xSemaphoreGive(connect_semaphore);
-
-    ESP_ERROR_CHECK(Initialize());
-
-    xTaskCreate(ConnectTask,
+    xTaskCreate(WifiTask,
         "wifi_connect_task",
         4096,
         (void*)this,
         12,
         &connect_task);
 }
 
+const std::vector<std::string>& Wifi::GetNetworksInRange() const
+{
+    return scanned_aps;
+}
+
+void Wifi::Connect(const std::string& ssid, const std::string& pwd)
+{
+    if (ssid.length() == 0 || pwd.length() == 0)
```

Comment:
```suggestion
    if (ssid.empty() || pwd.empty())
```
Prefer the more idiomatic way of saying 'empty'.